### DISCUSSION TITLE
Naming downloads with the version

### DISCUSSION THREAD

**Discussion by bigpresh** [GriefPrevention/GriefPrevention#1217] [Ideas]
When you download GP from the releases page, the file will always be named `GriefPrevention.jar`.

Would it be easy to have them named with the version, e.g. `GriefPrevention-16.7.1.jar`, so you can see at a glance what version you've got?  I always do it manually when downloading a build, but it would be awesome not to have to.

It would also make issues like #1216 easier to troubleshoot, as the error message when it fails to load the plugin would let you see from the filename what version it was trying to load.

There is, I suppose, an argument against it in case people get themselves in a mess by downloading a new GP version and dropping it in their `plugins/` folder without removing/renaming the old version, though.

**Comment by RoboMWM** [comment]
I prefer not having a version name appended, at least as part of the regular build process so it's easier for me to update jars on the server - I guess this is something that I could do as part of deploying to bukkitdev though.

**Reply by RoboMWM** [reply]
I plan to automate deployments to bukkitdev - they have an API, just haven't found the time to figure out how to either a) make a maven plugin to do this or b) make a github action to do this. https://authors.curseforge.com/knowledge-base/projects/529-api with tokens generated at https://dev.bukkit.org/account/api-tokens

**Reply by RoinujNosde** [reply]
There already is an [action](https://github.com/marketplace/actions/upload-to-curseforge) for that!
I'm using it in my project: https://github.com/RoinujNosde/SimpleClans/actions/runs/659210201/workflow

**Reply by Jikoo** [reply]
I'm also now successfully using the same action. Had loads of trouble initially with changelogs, not sure if it was on GH's end or Curse because I didn't heavily debug it. My version also uses markdown for the changelog because I like pretty things, bit of an added complication. Rather than recompile for Curseforge upload (as my compile depends on running BuildTools for various Spigot versions and as a result takes several minutes) mine pulls the binary directly from the Github release. Because the initial file attachment to the release is done automatically I never have to worry about whether the filename is correct.

My entire release process is now this:
* Run [a script](https://github.com/Jikoo/OpenInv/blob/3103e8822b77d2292c526266f623d42a045d908a/scripts/tag_release.sh) to increment and tag version
* Push commits and tag (could be done in the same script, but it's nice ensuring it compiles before pushing)
* Go get coffee while [an action](https://github.com/Jikoo/OpenInv/blob/3103e8822b77d2292c526266f623d42a045d908a/.github/workflows/ci.yml#L50-L77) generates a draft release with changelog and compiled binary attached (in my case, 2.5 minutes + 2.5 minutes per NMS revision or so)
* Edit the generated changelog to satisfaction
* Hit the release button
* Go do whatever else while [another action](https://github.com/Jikoo/OpenInv/blob/3103e8822b77d2292c526266f623d42a045d908a/.github/workflows/release.yml) clones the release to CurseForge 🎉 


End result: https://github.com/Jikoo/OpenInv/releases/tag/4.1.10 -> https://dev.bukkit.org/projects/openinv/files/3684690 (review pending so probably not visible yet)

**Reply by RoboMWM** [reply]
oooh, you have an action to publish to curseforge, neato

**Comment by RoboMWM** [comment]
That being said idk if that'll resolve anything, as the version alone on the jar doesn't indicate compatibilities anyways - a blind download will occur regardless in those support situations.

If anything, a command to print out debug information to paste in an issue would probably be the way to go.

**Reply by bigpresh** [reply]
Oh, sure, users will still blindly download it (probably even if it had a long unwieldy name like `GriefPrevention_16.7.1_for_MC_1.16.x.jar`!) but at least we could see at a glance what version they were trying to load.

A command to print debug info to stick in a ticket would be excellent... but that won't work so well when the plugin couldn't be loaded because it was for a newer version of MC than you have :)

I don't feel particularly strongly either way on this, I just think it has that benefit of making it clear what version they were trying to load, and makes my life that bit easier (I like being able to see at a glance from a .jar what version it is), but I can always just rename it upon download as I do at present with GP and plenty of others.

**Reply by RoboMWM** [reply]
Oh I see, it does show up in the logs ok.

**Reply by RoboMWM** [reply]
well, sure, that's a valid use case then for uploading to bukkitdev
